### DISCUSSION TITLE
SUID not working with pyinstaller-derived binaries?

### DISCUSSION THREAD

**Discussion by Fairlight2cx** [pyinstaller/pyinstaller#5539] [Help] [answered]
I just created a binary with pyinstaller.  The program in question is not reading as UID 0 via os.getuid() or os.geteuid(), despite the fact it is 4711 root:root.

Is there a way to get the resultant binaries to honour their SUID bit?  They execute, but the SUID bit appears to be being dropped on the floor.  Yet...they're not scripts.

I use StaticPerl for this all the time for Perl.  I have to believe pyinstaller can do it for Python.

A tiny bit of help, please?

**Comment by rokm** [comment, marked as answer]
Seems to work for me:
```python
# program.py
import os
print(f"UID: {os.getuid()}")
print(f"EUID: {os.geteuid()}")
```

```
(venv) [rok@alpha pyi-suid]$ pyinstaller program.py --clean --noconfirm --onefile
...
(venv) [rok@alpha pyi-suid]$ ls -l dist/program 
-rwxr-xr-x. 1 rok rok 7241368 feb  8 22:23 dist/program
(venv) [rok@alpha pyi-suid]$ stat -c "%a" dist/program 
755
(venv) [rok@alpha pyi-suid]$ ./dist/program 
UID: 1000
EUID: 1000

(venv) [rok@alpha pyi-suid]$ sudo chown root:root dist/program 
(venv) [rok@alpha pyi-suid]$ sudo chmod 4711 dist/program
(venv) [rok@alpha pyi-suid]$ stat -c "%a" dist/program 
4711
(venv) [rok@alpha pyi-suid]$ ls -l dist/program 
-rws--x--x. 1 root root 7241368 feb  8 22:23 dist/program
(venv) [rok@alpha pyi-suid]$ ./dist/program 
UID: 1000
EUID: 0
```

(Same for `onedir` build).

**Reply by Fairlight2cx** [reply]
Hmmm.  Now it's inexplicably working.  Okayyyy...  Sorry to have bothered you, and thanks for the sanity check!
